dm/dm/master/server.go

```diff
@@ -1070,6 +1068,46 @@ func (s *Server) getStatusFromWorkers(ctx context.Context, sources []string, tas
 	}
 	wg.Wait()
 	s.fillUnsyncedStatus(workerResps)
+
+	getSubTaskStatusCountInResp := func(taskName string) int {
+		cnt := 0
+		for _, resp := range workerResps {
+			for _, status := range resp.SubTaskStatus {
+				if status.Name == taskName {
+					cnt += 1
+				}
+			}
+		}
+		return cnt
+	}
+	appendFakeResp := func(taskName, sourceName, msg string, cnt int) {
+		for i := 0; i < cnt; i++ {
```

Comment:
we should have **one** `QueryStatusResponse` with many fake `SubTaskStatus`

Reply:
thanks for remind me this 🧡🧡🧡🧡 , fix this in https://github.com/pingcap/ticdc/pull/3358/commits/f9ca67fbabfd9152794ba5206fdd1db083873559
now we the resp from dmctl look like this:

```
➜ dmctl query-status
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "",
            "sourceStatus": {
                "source": "mysql-replica-01",
                "worker": "worker1",
                "result": null,
                "relayStatus": null
            },
            "subTaskStatus": [
                {
                    "name": "test",
                    "stage": "Running",
                    "unit": "Sync",
                    "result": null,
                    "unresolvedDDLLockID": "",
                    "sync": {
                        "totalEvents": "27",
                        "totalTps": "0",
                        "recentTps": "0",
                        "masterBinlog": "(mysql-bin.000002, 5693)",
                        "masterBinlogGtid": "cd3cf208-edc0-11eb-852c-d674c270ea47:1-26",
                        "syncerBinlog": "(mysql-bin.000002, 5387)",
                        "syncerBinlogGtid": "",
                        "blockingDDLs": [
                        ],
                        "unresolvedGroups": [
                        ],
                        "synced": false,
                        "binlogType": "remote",
                        "secondsBehindMaster": "0"
                    }
                }
            ]
        },
        {
            "result": false,
            "msg": "can't find task=test2 from dm-worker for source=mysql-replica-02, please use dmctl list-member to check if worker is offline.",
            "sourceStatus": {
                "source": "mysql-replica-02",
                "worker": "source not bound",
                "result": null,
                "relayStatus": null
            },
            "subTaskStatus": [
                {
                    "name": "test2",
                    "stage": "InvalidStage",
                    "unit": "InvalidUnit",
                    "result": null,
                    "unresolvedDDLLockID": ""
                }
            ]
        }
    ]
}
```